### PR TITLE
LF: utility function for Canton to handle interfaces

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala [hunk 1]
```diff
@@ -79,4 +79,27 @@ object Blinding {
...
+        case action: Node.LeafOnlyAction =>
+          toEntries(action.templateId)
+        case exe: Node.Exercise =>
+          toEntries(exe.templateId) ++ exe.interfaceId.view.flatMap(toEntries)
+        case _: Node.Rollback =>
+          Iterable.empty
+      }
+    }
+    Relation.from(entries)
+  }
```
Comment (remyhaemmerle-da) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:100]: I will add tests in a subsequent PR, to unblock @rgugliel-da

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala [hunk 2]
```diff
@@ -79,4 +79,27 @@ object Blinding {
       nodes = filteredNodes,
     )
   }
+
+  /* Calculate the packages needed by a party to interpret the projection   */
+  private[engine] def partyPackages(
+      tx: VersionedTransaction,
+      bindingInfo: BlindingInfo,
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:86]:
```suggestion
      blindingInfo: BlindingInfo,
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala [hunk 3]
```diff
@@ -79,4 +79,27 @@ object Blinding {
       nodes = filteredNodes,
     )
   }
+
+  /* Calculate the packages needed by a party to interpret the projection   */
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:83]: move the docs to the method that is public

**Review by cocreature — APPROVED** [review]
thx!

**Review by rgugliel-da — APPROVED** [review]
Thanks!

**Review by andreaslochbihler-da — COMMENTED** [review]
LGTM. We'll have to extend this function when we add new interface features, but only when they have been implemented.

**Pull request merged**